### DISCUSSION TITLE
Why is test_inline! macro trying to read a file

### DISCUSSION THREAD

**Discussion by ptazithos** [swc-project/swc#8568] [Q&A] [answered]
I guess the test_inline! macro is supposed to compare an inline input with an inline output. However the actual implementation compares an inline input with file content.

Given the swc plugin template, there is a test at the bottom of lib.rs
```rust
test_inline!(
    Default::default(),
    |_| as_folder(TransformVisitor),
    boo,
    // Input codes
    r#"foo === bar;"#,
    // Output codes after transformed with plugin
    r#"foo == bar;"#
);
```
The test_line! macro invokes the function `test_inline_input_output`.
https://github.com/swc-project/swc/blob/fd7cf4c30a7b12e6f62c154ddb2fd05ca193f8d5/crates/swc_ecma_transforms_testing/src/lib.rs#L461-L477
then the function `read_to_string` inside `test_inline_input_output` takes the output string `r#"foo == bar;"#` as a file path to read a local file.
https://github.com/swc-project/swc/blob/fd7cf4c30a7b12e6f62c154ddb2fd05ca193f8d5/crates/swc_ecma_transforms_testing/src/lib.rs#L364

**Comment by kdy1** [comment, marked as answer]
It looks like my mistake. I'll fix it soon

**Reply by ptazithos** [reply]
Thanks!
